### PR TITLE
STORM-951. Storm Hive connector leaking connections.

### PR DESCRIPTION
**Pull request by harshach**

### CODE REVIEW

**Comment by Parth-Brahmbhatt** [comment]
+1.

**Comment by vesense** [comment]
+1 LGTM

File: external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java [hunk 1]
```diff
@@ -170,10 +170,10 @@ public void run() {
         }
     }
 
-    private void flushAllWriters()
+    private void flushAllWriters(boolean rollToNext)
         throws HiveWriter.CommitFailure, HiveWriter.TxnBatchFailure, HiveWriter.TxnFailure, InterruptedException {
         for(HiveWriter writer: allWriters.values()) {
-            writer.flush(true);
+            writer.flush(rollToNext);
```
Comment (caofangkun) [inline external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java]: If ```rollToNext == false``` should we ``` writer.close()``` ? 
Like:
```
 writer.flush(rollToNext);
 if (!rollToNext) {
    writer.close();
 }
```

Comment (harshach) [inline external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java]: @caofangkun closing Writer handled in different method you can flushAllAndCloseWriters().

**Pull request merged**